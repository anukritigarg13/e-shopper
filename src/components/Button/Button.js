import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Button = (props) => (
  <div>
    <Link to={props.link}><button type="button">{props.buttonName}</button></Link>
  </div>
);
Button.propTypes = {
  link: PropTypes.string.isRequired,
  buttonName: PropTypes.string.isRequired,
};
export default Button;
