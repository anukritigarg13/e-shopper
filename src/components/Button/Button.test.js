/* eslint-disable react/jsx-props-no-spreading */
import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import Button from './Button';

describe('Button component', () => {
  afterEach(() => jest.clearAllMocks);
  const mockProps = {
    link: '/mock',
    buttonName: 'Mock button',
  };
  it('should display button', () => {
    render(<BrowserRouter><Button {...mockProps} /></BrowserRouter>);
    const buttonElement = screen.getByText('Mock button');
    expect(buttonElement.tagName).toBe('BUTTON');
  });
  it('should link to /mock', () => {
    render(<BrowserRouter><Button {...mockProps} /></BrowserRouter>);
    const buttonElement = screen.getByText('Mock button');
    fireEvent.click(buttonElement);
    expect(document.location.href).toBe('http://localhost/mock');
  });
  it('should match snapshot', () => {
    const { container } = render(<BrowserRouter><Button {...mockProps} /></BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
});
