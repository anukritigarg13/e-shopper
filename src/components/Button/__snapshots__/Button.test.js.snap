// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<div>
  <div>
    <a
      href="/mock"
    >
      <button
        type="button"
      >
        Mock button
      </button>
    </a>
  </div>
</div>
`;
